edgexfoundry/edgex-go
# refactor!: Rework Core Commands via messaging to use new MessageBus Request API

BREAKING CHANGE: Topics configuration for Core Command has changed. Also the internal response topic is always `edgex/response/<responding-service-name>/<request-id>`. 

closes #4309

signed-off-by: Leonard Goodell <leonard.goodell@intel.com>







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  **N/A** Docs focus on external usage. This PR is just for internal usage.

## Testing Instructions
Run non-secure EdgeX stack w/o any device service
Stop Core Command
Build local Device Simple from this [PR's branch](https://github.com/edgexfoundry/device-sdk-go/pull/1305)
Build local App Service Template from this [PR's branch](https://github.com/edgexfoundry/app-functions-sdk-go/pull/1290)
Build Core Command locally from this branch
Run Core Command (`./core-command`)
Run Device Simple (`device-simple) twice so it properly finds device, etc. on the 2nd run
Run App Template (`./app-new-service`)
Verify App template log shows messages like:
```
level=DEBUG ts=2023-02-06T21:32:14.6025413Z app=app-new-service source=sample.go:116 msg="Issuing Command Query for device Simple-Device01 in pip
eline 'default-pipeline'"
level=DEBUG ts=2023-02-06T21:32:14.6066121Z app=app-new-service source=sample.go:126 msg="Device Simple-Device01 has 7 commands to choose from. (
in pipeline 'default-pipeline')"
level=DEBUG ts=2023-02-06T21:32:14.606666Z app=app-new-service source=sample.go:142 msg="Issuing Command Rotation for device Simple-Device01 in i
n pipeline 'default-pipeline'"
level=DEBUG ts=2023-02-06T21:32:14.6123333Z app=app-new-service source=sample.go:150 msg="SendGetCommand successfully received new event from GET
 command Rotation on Simple-Device01 device in pipeline 'default-pipeline'"
```
Let run for 10 mins and verify no time out errors occur.



## New Dependency Instructions (If applicable)
